operators/pkg/controller/elasticsearch/remotecluster/elasticsearch.go

```diff
@@ -75,7 +75,11 @@ func UpdateRemoteCluster(
 		}
 	}
 	// Update state
-	reconcileState.UpdateRemoteClusters(currentRemoteClusters)
+	// If the map is empty then it must be stored as nil, not an empty map. It is retrieved as a nil value from the API
+	// and it could lead to some reconcile loops that never converge on Openshift.
```

Comment:
I find the comment a bit misleading. We say `it must be stored as nil`, but that's not what we actually do.
What we do here IIUC is update the state only if there are some remote clusters. We ignore all cases where the map is either nil or empty.

What if we have 3 remote clusters at some point, then do some updates and get to 0 remote clusters: does it mean the reconcileState never gets updated to match we now have 0 remote clusters?